contrib/automation_tests/test_cases/connection_window.py

```diff
@@ -54,9 +54,15 @@ class FilterAndSelectFirstProcess(E2ETestCase):
     Select the first process in the process list and verify there is at least one entry in the list
     """
     def _execute(self, process_filter):
-        filter_edit = self.find_control('Edit', 'FilterProcesses')
```

Comment:
You could do the same using `wait_for_condition`, which also re-tries after an exception.

Reply:
Sorry, I have no idea how to do this. `wait_for_condition` takes a lambda, right? How would it be possible to extract the return value?

Reply:
Good point, I just realized it's not that straightforward in Python and not even possible before 3.8.
It's doable like this: https://stackoverflow.com/questions/6282042/assignment-inside-lambda-expression-in-python

But this might make it less readable and enforces Python 3.8, so I'm open to leaving it as is.

Reply:
Okay, I'll leave it as this then.